app/models/king.rb

```diff
@@ -2,10 +2,39 @@ class King < ChessPiece
   X_MOVE_FACTOR = 1
   Y_MOVE_FACTOR = 1
 
+  BLACK_QUEENSIDE_CASTLE_TRAVERSED_SPACES = [Coordinates.new(3, 8), Coordinates.new(4, 8)]
+  BLACK_KINGSIDE_CASTLE_TRAVERSED_SPACES  = [Coordinates.new(6, 8), Coordinates.new(7, 8)]
+  WHITE_QUEENSIDE_CASTLE_TRAVERSED_SPACES = [Coordinates.new(3, 1), Coordinates.new(4, 1)]
+  WHITE_KINGSIDE_CASTLE_TRAVERSED_SPACES  = [Coordinates.new(6, 1), Coordinates.new(7, 1)]
+
   def valid_move?(coordinates)
     return false if out_of_bounds?(coordinates)
     return false if diff_in_x(coordinates.x) > X_MOVE_FACTOR
     return false if diff_in_y(coordinates.y) > Y_MOVE_FACTOR
     !obstructed?(coordinates)
   end
+
+  def can_castle?(rook)
+    return false if rook.moved?
+    return false if moved?
+    return false if obstructed?(rook.coordinates)
+
+    free_castling_spaces(color).each do |space|
+      return false if king_capturable_at_space?(space)
+    end
+
+    true
+  end
+
+  def free_castling_spaces(color)
```

Comment:
Unused method argument - `color`. If it's necessary, use `_` or `_color` as an argument name to indicate that it won't be used. You can also write as `free_castling_spaces(*)` if you want the method to accept any arguments but don't care about them.